src/main/kotlin/com/terraformation/backend/customer/api/UsersController.kt

```diff
@@ -104,6 +99,38 @@ class UsersController(private val userStore: UserStore) {
     userStore.updatePreferences(payload.organizationId, payload.preferences)
     return SimpleSuccessResponsePayload()
   }
+
+  @ApiResponse200
+  @ApiResponse404
+  @GetMapping
+  @Operation(summary = "Gets a user by some criteria, for now only email is available")
+  fun searchUsers(
+      @RequestParam
+      @Schema(description = "The email to use when searching for a user")
+      email: String
+  ): GetUserResponsePayload {
+    val user = userStore.fetchByEmail(email)
```

Comment:
I think we need a wrapper function for this in `UserStore` that checks permissions. The `fetchByEmail` function is used internally in the auth code so it can't check permissions all the time (it's called in contexts where there's no "current user" yet, I believe) but we need to not allow random users to look up other random users' information.

For now, just checking for a global role is probably okay here. A more sophisticated permission check could also allow looking up people who are in one of the current user's organizations.